model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remote/RemoteUserSessionProvider.java

```diff
@@ -462,9 +461,10 @@ private Stream<UserSessionModel> streamUserSessionByClientId(RealmModel realm, S
         var userSessionTx = getUserSessionTransaction(offline);
         return Flowable.fromIterable(QueryHelper.toCollection(userSessionIdQuery, QueryHelper.SINGLE_PROJECTION_TO_STRING))
                 .flatMapMaybe(userSessionTx::maybeGet, false, MAX_CONCURRENT_REQUESTS)
-                .concatMapMaybe(session -> maybeInitUserSessionFromQuery(session, realm, offline))
+                .blockingStream()
+                .flatMap(session -> maybeInitUserSessionFromQuery(session, realm, offline))
```

Comment:
RxJava pipelines do not allow `null` but the Java `Stream` does. You can use the `initUserSessionFromQuery` and filter by non null. 
The method `maybeInitUserSessionFromQuery` can be removed. 
```suggestion
                .map(session -> initUserSessionFromQuery(session, realm, null, offline))
                .filter(Objects::nonNull)
```